### PR TITLE
FIX cache update logic

### PR DESCRIPTION
**Pull request by fgalan**
Issue #4069 and #4063

- [x] Review other issues solved by this PR
- [x] CNR
- [x] Review Devel Manual
  * Specially, looks for occurrences of mongoSubCountersUpdate() (in .md and diagrams) as that function as unfold into two separate functions: mongoSubUpdateOnNotif() and mongoSubUpdateOnCacheSync()

### CODE REVIEW

File: test/functionalTest/harnessFunctions.sh [hunk 1]
```diff
@@ -378,6 +378,12 @@ function localBrokerStart()
   then
     port=$CB_PORT
     CB_START_CMD="$CB_START_CMD_PREFIX -port $CB_PORT  -pidpath $CB_PID_FILE  -dbhost $dbHost:$dbPort -db $CB_DB_NAME -dbPoolSize $POOL_SIZE -t $traceLevels $IPvOption $extraParams"
+  elif [ "$role" == "CBHA" ]
```
Comment (fgalan) [inline test/functionalTest/harnessFunctions.sh:381]: To implement the test case for this PR (test/functionalTest/cases/4069_status_inactive_not_working_with_cache_in_ha/status_inactive_not_working_with_cache_in_ha.test) we have needed to extend the testing framework with a new role: CB for HA testing. It's like the usual CB but sharing the same database.

Thus, new HA cases can be tested from now on :)

NTC (informative)

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Comment by cesarjorgemartinez** [comment]
LGTM

**Comment by fgalan** [comment]
@fisuda this PR does some modification in .md documents. It would be great if you could synchronize in the Japanese translation, please. Thanks!

**Comment by fisuda** [comment]
I sent the PR #4075.
Thanks.

**Pull request merged**